src/py_zfs_pool.c

```diff
@@ -28,378 +24,251 @@ int py_zfs_pool_init(PyObject *type, PyObject *args, PyObject *kwds) {
 
 static
 void py_zfs_pool_dealloc(py_zfs_pool_t *self) {
+	if (self->zhp != NULL) {
+		Py_BEGIN_ALLOW_THREADS
+		zpool_close(self->zhp);
+		Py_END_ALLOW_THREADS
+		self->zhp = NULL;
+	}
+	Py_CLEAR(self->name);
+	Py_CLEAR(self->pylibzfsp);
 	Py_TYPE(self)->tp_free((PyObject *)self);
 }
 
 static
-PyObject *py_zfs_pool_asdict(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_reiterate_vdevs(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_unsup_features(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_wait(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
+PyObject *py_zfs_pool_get_name(py_zfs_pool_t *self, void *extra) {
+	if (self == NULL)
+		Py_RETURN_NONE;
+	return Py_NewRef(self->name);
 }
 
 static
-PyObject *py_zfs_pool_sync(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_create(PyObject *self, PyObject *args) {
+PyObject *py_zfs_pool_asdict(PyObject *self, PyObject *args) {
 	Py_RETURN_NONE;
 }
 
 static
-PyObject *py_zfs_pool_attach_vdevs(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
+PyObject *py_zfs_pool_root_dataset(PyObject *self, PyObject *args) {
+	py_zfs_dataset_t *out = NULL;
+	py_zfs_pool_t *p = (py_zfs_pool_t *)self;
+	PyObject *fargs = NULL;
+	PyObject *fkwargs = NULL;;
 
-static
-PyObject *py_zfs_pool_sync_vdev_by_guid(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
+	fargs = PyTuple_New(0);
+	if (fargs == NULL)
+		return (NULL);
 
-static
-PyObject *py_zfs_pool_delete(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
+	fkwargs = Py_BuildValue("{s:O}", "name", p->name);
+	if (fkwargs == NULL) {
+		Py_DECREF(fargs);
+		return (NULL);
+	}
 
-static
-PyObject *py_zfs_pool_start_scrub(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
+	out = (py_zfs_dataset_t *)py_zfs_resource_open((PyObject *)p->pylibzfsp,
```

Comment:
I believe the more correct way to do this would be to get the method object for the call to `open_resource`:
```
PyObject *open_rsrc = PyObject_GetAttrString((PyObject*)p->pylibzfsp, "open_resource");
```

Then once you have it, you can call via python:
```
out = PyObject_Call(open_rsrc, fargs, fkwargs);
```

This avoids having to expose py_zfs_resource_open in pylibzfs2.h.